File: backend/app/schemas/send_log.py
from datetime import datetime

from pydantic import BaseModel


class SendLogRead(BaseModel):
    id: int
    campaign_id: int
    contact_id: int
    contact_email: str | None = None
    mailbox_id: int | None
    mailbox_email: str | None = None
    subject_used: str
    body_preview: str | None
    status: str
    error_message: str | None
    error_label: str | None = None
    sent_at: datetime

    model_config = {"from_attributes": True}


class SendLogFilter(BaseModel):
    mailbox_id: int | None = None
    status: str | None = None

File: frontend/src/types/sendLog.ts
export interface SendLog {
  id: number;
  campaign_id: number;
  contact_id: number;
  contact_email: string | null;
  mailbox_id: number | null;
  mailbox_email: string | null;
  subject_used: string;
  body_preview: string | null;
  status: string;
  error_message: string | null;
  error_label: string | null;
  sent_at: string;
}
